File: spl-net/src/main/java/bgu/spl/net/impl/messages/AckMessage.java
package bgu.spl.net.impl.messages;

public class AckMessage extends Message {

    private short messageOpcode;

    public AckMessage(short messageOpcode)
    {
        super((short)10);
        this.messageOpcode=messageOpcode;
    }

    public short getMessageOpcode(){return messageOpcode;}
    public String toString(){
        short opcode=getOpCode();
        String output=String.valueOf(opcode);
        short messageOpcode=getMessageOpcode();
        output+=messageOpcode;
        return output;
    }
}

File: Boost_Echo_Client/src/AckMessage.cpp
//
// Created by MAY on 04/01/2022.
//

#include "../include/AckMessage.h"

AckMessage::AckMessage(short _messageOpcode, short _opCode) : Message(_opCode), messageOpcode(_messageOpcode) {
}

short AckMessage::getMessageOpCode() {
    return messageOpcode;
}

std::string AckMessage::toString() {
    std::string answer = "ACK ";
    answer.append(std::to_string(getMessageOpCode()));
    return answer;
}
